OpenRA.Mods.Common/Traits/CustomOwnerLostAction.cs

```diff
@@ -0,0 +1,31 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public enum OwnerLostAction { ChangeOwner, Dispose, Nothing }
```

Comment:
I would drop Nothing altogether, that can be abused to post-defeat units and i don't think we should have an option for such.

Reply:
I kind of agree here, but as much as i'd like to restrict that I would like more to have streamlined code.

IMO it would be better to drop the `Custom` in the trait name here, change `Nothing` to `Kill`, and then remove the "if actor doesn't have this trait then kill it" logic from the *VictoryConditions traits.  This trait can then be added to the defaults for all player-built actors.

Reply:
Note that this will need a placeholder update rule that explains that `OwnerLostAction` now needs to be defined on all player-built actors.